payjoin-cli/src/app/v2.rs

```diff
@@ -275,14 +275,16 @@ impl App {
         _ = try_contributing_inputs(&mut provisional_payjoin.inner, &bitcoind)
             .map_err(|e| log::warn!("Failed to contribute inputs: {}", e));
 
-        if !provisional_payjoin.is_output_substitution_disabled() {
-            // Substitute the receiver output address.
-            let receiver_substitute_address = bitcoind
-                .get_new_address(None, None)
-                .map_err(|e| Error::Server(e.into()))?
-                .assume_checked();
-            provisional_payjoin.substitute_output_address(receiver_substitute_address);
-        }
+        _ = provisional_payjoin
+            .try_substitute_receiver_output(|| {
+                Ok(bitcoind
+                    .get_new_address(None, None)
+                    .map_err(|e| Error::Server(e.into()))?
+                    .require_network(network)
+                    .map_err(|e| Error::Server(e.into()))?
+                    .script_pubkey())
+            })
+            .map_err(|e| log::warn!("Failed to substitute output: {}", e));
```

Comment:
V1 receivers are long running hosted servers and only output a single bip21 per run, it made sense to me to have them substitute outputs by default.

V2 receivers on the other hand are always session based and serve only one sender. It makes less sense to me for them to try to substitute outputs. I understand that your change is unrelated but imo v2 receivers should be changed not to substitute addresses by default in a follow up.

Reply:
Removed in v2 receiver